### PR TITLE
index-edit-delete team

### PR DESCRIPTION
**Pull request by phuong1492**

### CODE REVIEW

File: app/controllers/admin/teams_controller.rb [hunk 1]
```diff
@@ -12,7 +12,7 @@ def show
 
   def new
     @team = Team.new
-    @leaders = User.user_not_in_team
+    @free_user = User.user_not_in_team
```
Comment (DucNguyenVan) [inline app/controllers/admin/teams_controller.rb]: @free_users

File: app/controllers/admin/teams_controller.rb [hunk 2]
```diff
@@ -26,6 +26,28 @@ def create
...
+  def edit
+    @free_user = User.user_not_in_team
+    @users_in_team = @team.users.pluck :name, :id
+    @leaders = @free_user + @team.users
+  end
+
+  def update
+    if @team.update_attributes team_params
+      check_leader
+      flash[:success] = "Success"
```
Comment (phonghenry14) [inline app/controllers/admin/teams_controller.rb]: i18n

File: app/views/admin/teams/_team.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<li>
+  <div class= "col-md-3 col-md-offset-1">
+    <%= link_to team.name, [:admin, team] %></div>
+  <div class= "col-md-3 col-md-offset-5">
+    <%= link_to t("admin.team.detail"), [:admin, team], class: "btn btn-primary" %>
+    <%= link_to t("admin.team.edit"), [:edit, :admin, team], class: "btn btn-primary" %>
+    <%= link_to t("admin.team.delete"), [:admin, team], data: {confirm: t("admin.team.confirm")}, 
+      method: :delete, class: "btn btn-primary" %>
+  </div>
+</li>
```
Comment (phonghenry14) [inline app/views/admin/teams/_team.html.erb]: xuống dòng

File: app/controllers/admin/teams_controller.rb (hunk 1 above)
Comment (DucNguyenVan) [inline app/controllers/admin/teams_controller.rb]: @free_users

File: app/views/admin/teams/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  <%= f.label :name %>
+  <%= f.text_field :name, class: "form-control" %>
+
+  <%= f.label :description %>
+  <%= f.text_field :description, class: "form-control" %>
+
+  <%= f.label :leader %>
+  <%= f.collection_select :leader_id, @leaders, :id, :name %>
+  <div class = "row">
+	  <%= render "users_in_team", f: f %>
```
Comment (phonghenry14) [inline app/views/admin/teams/edit.html.erb]: indent

File: app/views/admin/teams/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+	  <%= render "users_not_in_team", f: f %>
+  </div>
+  <br />
+  <p>
+	  <%= f.submit :submit, class: "btn btn-primary col-md-12" %>
```
Comment (phonghenry14) [inline app/views/admin/teams/edit.html.erb]: ident

File: app/views/admin/teams/_form.html.erb [hunk 6]
```diff
@@ -1,21 +1,17 @@
 <%= form_for [:admin, @team] do |f| %>
+  
   <%= f.label :name %>
```
Comment (daothanhcam) [inline app/views/admin/teams/_form.html.erb]: i18n

File: app/views/admin/teams/_form.html.erb [hunk 7]
```diff
@@ -1,21 +1,17 @@
...
   <%= f.text_field :name, class: "form-control" %>
+  
   <%= f.label :description %>
```
Comment (daothanhcam) [inline app/views/admin/teams/_form.html.erb]: i18n

File: app/views/admin/teams/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  <%= f.label :description %>
+  <%= f.text_field :description, class: "form-control" %>
+
+  <%= f.label :leader %>
+  <%= f.collection_select :leader_id, @leaders, :id, :name %>
+  <div class = "row">
+	  <%= render "users_in_team", f: f %>
+	  <%= render "users_not_in_team", f: f %>
+  </div>
+  <br />
```
Comment (DucNguyenVan) [inline app/views/admin/teams/edit.html.erb]: should be convert to <br/>

File: app/views/admin/teams/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+<% provide :title, t("admin.team.all_team") %>
+<div class="row">
+  <div class="box">
+    <div class="col-lg-12">
+      <hr>
+      <h2 class="intro-text text-center"><strong><%= t "admin.team.all_team" %></strong></h2>
+      <hr>
+      <ul class="col-lg-12">
+        <div class="col-md-offset-9"><%= will_paginate %></div>
+          <%= render partial: "team", collection: @teams %>
```
Comment (DucNguyenVan) [inline app/views/admin/teams/index.html.erb]: you had _team.html.erb, so render should be replace by short hand :
<%= render @teams %>

File: app/views/admin/teams/_form.html.erb [hunk 10]
```diff
@@ -1,21 +1,17 @@
...
-        </div>       
-      <% end %>
-    </div>
-  </p>
+  <%= f.collection_select :leader_id, @free_users, :id, :name %>
+  
+  <div class = "row">
+    <%= render "users_not_in_team", f: f %>
+  </div>
   <br />
```
Comment (DucNguyenVan) [inline app/views/admin/teams/_form.html.erb]: remove space in br tag

File: app/views/admin/teams/_users_not_in_team.html.erb [hunk 11]
```diff
@@ -0,0 +1,17 @@
+<div class= "col-xs-6 ">
```
Comment (DucNguyenVan) [inline app/views/admin/teams/_users_not_in_team.html.erb]: indent

File: app/views/admin/teams/_team.html.erb [hunk 12]
```diff
@@ -0,0 +1,13 @@
...
+  <div class= "col-md-3 col-md-offset-5">
+    <%= link_to t("admin.team.detail"), [:admin, team],
+      class: "btn btn-primary" %>
+    <%= link_to t("admin.team.edit"), [:edit, :admin, team],
+      class: "btn btn-primary" %>
+    <%= link_to t("admin.team.delete"), [:admin, team],
+      data: {confirm: t("admin.team.confirm")}, 
+      method: :delete, class: "btn btn-primary" %>
+  </div>
+</li>
```
Comment (phonghenry14) [inline app/views/admin/teams/_team.html.erb]: xuống dòng này

File: app/views/admin/teams/_users_in_team.html.erb [hunk 13]
```diff
@@ -0,0 +1,17 @@
+<div class= "col-xs-6 ">
```
Comment (DucNguyenVan) [inline app/views/admin/teams/_users_in_team.html.erb]: indent

Comment (phonghenry14) [inline app/views/admin/teams/_users_in_team.html.erb]: indent

File: app/controllers/admin/teams_controller.rb (hunk 2 above)
Comment (DucNguyenVan) [inline app/controllers/admin/teams_controller.rb]: fuck this shjt

File: app/views/admin/shared/_error_messages.html.erb [hunk 14]
```diff
@@ -0,0 +1,12 @@
+<% if object.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize object.errors.count, "error" %>
+      <ul>
+        <% object.errors.full_messages.each do |msg|%>
+        <li><%= msg %></li>
```
Comment (phonghenry14) [inline app/views/admin/shared/_error_messages.html.erb]: indent

File: app/views/admin/teams/_users_in_team.html.erb [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+      <tr>
+        <th><%= t "admin.team.all_user" %></th>
+      </tr>
+    </thead>
+    <tbody>
+      <%= f.collection_check_boxes :user_ids, @team.users, :id, :name do |b| %>
+      <tr>
+        <td><%= b.check_box %>
+          <%= b.object.name %></td>
+        </tr>
```
Comment (phonghenry14) [inline app/views/admin/teams/_users_in_team.html.erb]: indent

File: app/views/admin/teams/_users_not_in_team.html.erb [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+      <tr>
+        <th><%= t "admin.team.add_user" %></th>
+      </tr>
+    </thead>
+    <tbody>
+      <%= f.collection_check_boxes :user_ids, @free_users, :id, :name do |b| %>
+      <tr>
+        <td><%= b.check_box %>
+          <%= b.object.name %></td>
+        </tr>
```
Comment (phonghenry14) [inline app/views/admin/teams/_users_not_in_team.html.erb]: idnet

File: app/views/admin/teams/_users_not_in_team.html.erb [hunk 17]
```diff
@@ -0,0 +1,17 @@
+<div class="col-xs-6">
+  <table class="table table-bordered">
+    <thead>
+      <tr>
+        <th><%= t "admin.team.add_user" %></th>
+      </tr>
+    </thead>
+    <tbody>
+      <%= f.collection_check_boxes :user_ids, @free_users, :id, :name do |b| %>
+      <tr>
```
Comment (phonghenry14) [inline app/views/admin/teams/_users_not_in_team.html.erb]: ident

File: app/views/admin/teams/_users_in_team.html.erb [hunk 18]
```diff
@@ -0,0 +1,17 @@
+<div class="col-xs-6">
+  <table class="table table-bordered">
+    <thead>
+      <tr>
+        <th><%= t "admin.team.all_user" %></th>
+      </tr>
+    </thead>
+    <tbody>
+      <%= f.collection_check_boxes :user_ids, @team.users, :id, :name do |b| %>
+      <tr>
```
Comment (phonghenry14) [inline app/views/admin/teams/_users_in_team.html.erb]: indent

File: app/controllers/admin/teams_controller.rb [hunk 19]
```diff
@@ -26,6 +26,28 @@ def create
...
+      check_leader
+      flash[:success] = t "admin.team.success"
+      redirect_to admin_team_path @team
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @team.destroy
```
Comment (khanhkhanh) [inline app/controllers/admin/teams_controller.rb]: if ... else ... end

File: app/views/admin/shared/_error_messages.html.erb [hunk 20]
```diff
@@ -0,0 +1,12 @@
+<% if object.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize object.errors.count, "error" %>
```
Comment (khanhkhanh) [inline app/views/admin/shared/_error_messages.html.erb]: If you have this file in app/views/shared, use it and remove this file

File: app/controllers/admin/teams_controller.rb [hunk 21]
```diff
@@ -35,4 +60,10 @@ def team_params
     params.require(:team).permit :name, :description, :leader_id,
                                  :team_id, user_ids: []
   end
+
+  def check_leader 
+    if Team.find_by(id: :leader_id).nil?
```
Comment (khanhkhanh) [inline app/controllers/admin/teams_controller.rb]: Try using @team.users << @team.leader unless Team.find leader_id

File: app/controllers/admin/teams_controller.rb [hunk 22]
```diff
@@ -12,7 +12,7 @@ def show
 
   def new
     @team = Team.new
-    @leaders = User.user_not_in_team
+    @free_users = User.user_not_in_team
```
Comment (dieunb) [inline app/controllers/admin/teams_controller.rb]: `user_not_in_team` naming is not good. Don't include model name in scope name

File: app/controllers/admin/teams_controller.rb [hunk 23]
```diff
@@ -35,4 +60,9 @@ def team_params
     params.require(:team).permit :name, :description, :leader_id,
                                  :team_id, user_ids: []
   end
+
+  def check_leader 
+    @team.users << @team.leader unless @team.users.
+      find_by(id: team_params[:leader_id])
```
Comment (dieunb) [inline app/controllers/admin/teams_controller.rb]: remove `()`

**Pull request merged**